### ISSUE TITLE
Wifi network list doesn't merge multiple APs with same SSID

### ISSUE THREAD

**Issue by a-kohout** [koreader/koreader#2639]
* KOReader version: v2015.11-896-g4316284
* Device: Kobo Glo HD

#### Issue
Due to the size of the house, there are two access points with the same name/security settings connected to the same network. Nickel and every other wifi capable device shows the network once since the operating system uses the access point with better reception automatically
KOReader shows both access points.
For multiple networks with the same SSID, only the strongest should be shown
![filemanager_2017-mar-12_231948](https://cloud.githubusercontent.com/assets/26367956/23837257/ac346490-0785-11e7-96e7-91f9a25203ff.png)

#### Steps to reproduce
Connect KOreader to multi-ap wifi network

**Comment by Hzj-jie** [comment]
I do not have too much knowledge about SSID, but is it a correct behavior just to deduplicate by text?

**Comment by KenMaltby** [comment]
Are you sure that it is not connecting to "the access point with better reception", it is listed first and shows the most bars.  Not sure what advantage there is in having the access point detection ignore the next closest wifi just because it has the same SSID.  You could try changing the SSIDs to Aloha1 & Aloha2

**Comment by Frenzie** [comment]
@Hzj-jie A quick search suggests that indeed it is: https://superuser.com/a/738541

Also note:

>  That is to say, I've seen 802.11 clients that were susceptible to security downgrade attacks, where just because they knew a WPA2-PSK password for a given SSID, didn't mean they felt compelled to always require that network to use WPA2-PSK; if they saw the same SSID without security, they might just join it and not bother with security. This is, of course, insecure and buggy behavior, and I would hope that the Wi-Fi Alliance's Wi-Fi certification testing tests to make sure that devices that pass certification don't do this, but you never know.

**Comment by a-kohout** [comment]
I just walked around the house, checking the status pages on each access point. The device indeed roams between APs as expected.
So wpa_supplicant already behaves as expected, only koreader's GUI doesn't
Renaming the SSIDs would be a step backwards in usability. 
I can walk upstairs without a download interrupting, as the system seamlessly roams between APs. With the SSIDs changed this wouldn't work seamlessly. This type of network installation is also standard in any business that is bigger than a one-room office.
It seems the strongest one shown on the network list page is indeed the closest one. The second one is the one further away.

If your neighbor's Wifi has the same name but a different password, then the system should ask for the password.
If your neighbor's wifi has different security settings, then macOS for example warns you, asking if you are sure you want to connect. Windows outright refuses to connect to a wifi with different security settings but the name of an existing connection, needing to delete it first.

A rogue AP with a different password in a multi-AP setup would thus cause a connection failure when the device tries to roam to that AP. Since authentication works on a 4-way handshake that rogue AP wouldn't be able to capture the password.

So a simple deduplication of the network list seems to be the proper way. Though only if the security settings are the same.

**Comment by a-kohout** [comment]
I just set security to none on one of the APs, to see what macOS would do. It simply ignores that AP, even though it was the stronger one. It doesn't list it separately either, just if it wasn't there.

**Comment by KenMaltby** [comment]
So "ignorance is bliss", I guess you could just ignore the "extra" listing?

**Comment by lgeek** [comment]
> If your neighbor's Wifi has the same name but a different password, then the system should ask for the password.

You'd have to explicitly try to connect to that AP and see that the 4 way handshake fails to tell that it uses a different password. It's also non-trivial to say whether the handshake failed because the password is wrong or some other reason (e.g. packet loss). At the moment, the network manager just throws a timeout error (after a pretty lengthy time blocking the UI).

**Comment by poire-z** [comment]
(Mentionning for reference a possibly related issue:
https://www.mobileread.com/forums/showthread.php?t=332055 )

**Comment by a-kohout** [comment]
The connection issue reported in the mobileread thread appears to be caused by #6500 

FWIW the SSID duplication issue is much worse in corporate environments. Here, there's 3 SSIDs broadcasting on every Access point (100 total).

Koreader shows each SSID from these APs 7 or 8 times, filling up two pages before I see my hotspot. It doesn't sort alphabetically. Nickel merges the SSIDs and shows 4.

Nickel also has an option to manually connect (to a hidden network).

**Comment by pazos** [comment]
@a-kohout (and others interested): feel free to give it a try. It shouldn't be too hard. The function that retrieves the network aps is https://github.com/koreader/koreader/blob/a2dcfe9aec58f734e1285d1b6ac5a13271275b91/frontend/ui/network/wpa_supplicant.lua#L17-L45

We stored retrieved aps in a table called `list`. We use the function `scanThenGetResults` from [lj-wpaclient]( https://github.com/koreader/lj-wpaclient/blob/880904033e4b145cbda72108c673decdd9d7be08/wpaclient.lua#L130-L158) module.

I would prefeer to parse and clean the `list` table on the frontend and leave lj-wpasupplicant as is, since it shows the same behaviour as wpa_cli.

**Comment by NiLuJe** [comment]
No, the thread predates FW 4.23, it's about wpa_supplicant doing weird things with multiples BSSIDs on the same SSIDs, and our client aborting too early on the first failure.
